.github/workflows/pr_testing_check.yaml

```
@@ -0,0 +1,119 @@
+name: Run testcases that changed on this PR
+
+on: pull_request
+
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v4.1.2
+
+    - name: Set up Python
+      id: setup-python
+      uses: actions/setup-python@v5
+      with:
+        python-version: '3.12'
+
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install pylint
+
+    - name: Run pre-commit
+      uses: pre-commit/action@v3.0.1
+
+  build-and-test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4.1.2
+
+
+      - name: Install OpenVPN
+        run: |
+          sudo apt update
+          sudo apt install -y openvpn openvpn-systemd-resolved
```

I'm not sure, but I think apt-get save the a cahce for the dependencies and is on `/var/cache/apt/archives` now, I'm not sure if you can follow the pattern on the cache action page and use something like `key: ${{ runner.os }}-openvpn-${{ hashFiles('**/apt.lock') }}` to create a key for future share.

```suggestion
      - name: Cache OpenVPN installation
        uses: actions/cache@vd
        with:
          path: /var/cache/apt/archives
          key: ${{ runner.os }}-openvpn-${{ hashFiles('**/apt.lock') }}
          restore-keys: |
            ${{ runner.os }}-openvpn-

      - name: Install OpenVPN
        run: |
          sudo apt update
          sudo apt install -y openvpn openvpn-systemd-resolved
```